.github/workflows/library_interop_tests.yml

```diff
@@ -0,0 +1,186 @@
+          path: ./${{matrix.library}}/runtimes/${{matrix.language}}/*.json
+
+  testInteroperablity:
+    needs: generateEncryptVectors
+    strategy:
+      matrix:
+        library: [TestVectorsAwsCryptographicMaterialProviders]
+        os: [
+            # TODO just test on mac and ubuntu for now
+            # windows-latest,
+            ubuntu-latest,
+            macos-latest,
+          ]
+        encrypting_language: [java, net]
+        decrypting_language: [java, net]
+        dotnet-version: ["6.0.x"]
+    runs-on: ${{ matrix.os }}
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Support longpaths on Git checkout
+        run: |
+          git config --global core.longpaths true
+      # KMS and MPL tests need to use credentials which can call KMS
+      - name: Configure AWS Credentials for Tests
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-region: us-west-2
+          role-to-assume: arn:aws:iam::370957321024:role/GitHub-CI-MPL-Dafny-Role-us-west-2
+          role-session-name: InterOpTests
+
+      - uses: actions/checkout@v3
+      # The specification submodule is private so we don't have access, but we don't need
+      # it to verify the Dafny code. Instead we manually pull the submodule we DO need.
+      - run: git submodule update --init libraries
+
+      # Set up runtimes
+      - name: Setup .NET Core SDK ${{ matrix.dotnet-version }}
+        if: matrix.decrypting_language == 'net'
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: ${{ matrix.dotnet-version }}
+
+      - name: Setup Java 17
+        if: matrix.decrypting_language == 'java'
+        uses: actions/setup-java@v3
+        with:
+          distribution: "corretto"
+          java-version: 17
+
+      - name: Setup Dafny
+        uses: dafny-lang/setup-dafny-action@v1.6.1
+        with:
+          dafny-version: ${{ inputs.dafny }}
+
+      # Build implementation for each runtime
+      - name: Build ${{ matrix.library }} implementation in Java
+        if: matrix.decrypting_language == 'java'
+        working-directory: ./${{ matrix.library }}
+        shell: bash
+        run: |
+          # This works because `node` is installed by default on GHA runners
+          CORES=$(node -e 'console.log(os.cpus().length)')
+          make build_java CORES=$CORES
+
+      - name: Build ${{ matrix.library }} implementation in .NET
+        if: matrix.decrypting_language == 'net'
+        working-directory: ./${{ matrix.library }}
+        shell: bash
+        run: |
+          # This works because `node` is installed by default on GHA runners
+          CORES=$(node -e 'console.log(os.cpus().length)')
+          make transpile_net
+
+      - name: Download Encrypt Manifest Artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{matrix.os}}_vector_artifact_${{matrix.encrypting_language}}
```

Comment:
I think we also need to distinguish between encrypting OS/decrypting OS

Reply:
Maybe we don't need this now since ubuntu and mac are virtually the same.
Pros and cons around differentiating encrypting/decrypting by OS. Pros: better test coverage. Cons: too many tests as we add more languages/OSes (flaky tests, etc)

Reply:
I vote we cover OS-es ONLY via Unit Tests/package specific tests.
Not integ/inter-op tests. 
We SHOULD validate that our  Unit Tests/package specific tests coverage of Crypto Primitives covers AES-GCM, RSA, and any other primitive in a round trip. 
But our test coverage is pretty exhaustive already...